shared/bettersystem/bettersystem.h

```diff
@@ -39,3 +42,16 @@ int e_execsp(std::string cmd, std::string cat1, std::string cat2, std::string pa
 int e_execsp(std::string cmd, std::string cat1, std::string cat2, std::string cat3, std::string path);
 
 //int better_system(std::string,std::string,std::string="",const char* = NULL);
+
+inline std::string string_replace_all(std::string str, const std::string& substr, const std::string& nstr)
```

Comment:
Josh, did you miss where I said:
>Cleaned up a few redundant string_replace_all implementations

This is actually 1 less `string_replace_all` I moved it to better system so it's shared. Now what we could do is move it from bettersystem to its own shared string library and delete the redundant `estring.h` version of it too.

That said, yeah I changed it to return a copy of the string instead of mutating. It seems we also happen to have two versions in our source code. Which do you prefer, one that mutates the original string or one that returns a mutated copy of the string?